tests/test_multi_ai_cache_privileges.py

```
@@ -0,0 +1,77 @@
+
+def test_multi_ai_cache_privileges_enforcement(tmp_path, monkeypatch):
+    test_cache_dir = tmp_path / ".multi-ai-cache"
+
+    import core.cache as mc
+    monkeypatch.setattr(mc, "CACHE_DIR", test_cache_dir)
+
+    test_cache_dir.mkdir(parents=True, exist_ok=True)
+    if not test_cache_dir.is_symlink():
+        try:
+            test_cache_dir.chmod(0o700)
+        except Exception:
```

Pylint (4.0.7)

[warning] Catching too general exception Exception

(W0718)